### PR TITLE
[test/sink] fix tensor-config in test 

### PR DESCRIPTION
**Pull request by kimjh12**
- Fix unittest_sink & unittest_plugin
- fix merge & converter test

**Self evaluation:**
1. Build test: [*]Passed [ ]Failed [ ]Skipped
2. Run test: [*]Passed [ ]Failed [ ]Skipped

### CODE REVIEW

File: gst/nnstreamer/tensor_common.c [hunk 1]
```diff
@@ -1235,6 +1235,48 @@ gst_tensors_caps_from_config (const GstTensorsConfig * config)
...
+    c.rate_d = config->rate_d;
+
+    gst_caps_append (caps, gst_tensor_caps_from_config (&c));
+  }
+
+  /* other/tensors */
+  gst_caps_append (caps, gst_tensors_caps_from_config (config));
+
+  /* other/tensors-flexible */
+  if (!gst_tensors_info_is_flexible (&config->info)) {
```
Comment (gichan-jang) [inline gst/nnstreamer/tensor_common.c:1268]: Is it correct?
```suggestion
  if (gst_tensors_info_is_flexible (&config->info)) {
```

Comment (kimjh12) [reply gst/nnstreamer/tensor_common.c:1268]: Yes. Same code copied from `tensor_filter.c` which always appends flexible caps

Comment (gichan-jang) [reply gst/nnstreamer/tensor_common.c:1268]: Ah.. Thanks for the explanation.

File: tests/nnstreamer_sink/unittest_sink.cc [hunk 2]
```diff
@@ -2427,21 +2338,18 @@ TEST (tensorStreamTest, audioS8)
...
   EXPECT_FALSE (g_test_data.invalid_timestamp);
 
   /** check tensor config for audio */
-  EXPECT_TRUE (gst_tensor_config_validate (&g_test_data.tensor_config));
-  EXPECT_EQ (g_test_data.tensor_config.info.type, _NNS_INT8);
-  EXPECT_EQ (g_test_data.tensor_config.info.dimension[0], 1U);
-  EXPECT_EQ (g_test_data.tensor_config.info.dimension[1], 500U);
-  EXPECT_EQ (g_test_data.tensor_config.info.dimension[2], 1U);
-  EXPECT_EQ (g_test_data.tensor_config.info.dimension[3], 1U);
-  EXPECT_EQ (g_test_data.tensor_config.rate_n, 16000);
```
Comment (anyj0527) [inline tests/nnstreamer_sink/unittest_sink.cc:2443]: gtest will complain about the sign v.s. unsigned compare.
`16000U -> 16000`
`1U -> 1`

File: tests/nnstreamer_sink/unittest_sink.cc [hunk 3]
```diff
@@ -2559,21 +2461,18 @@ TEST (tensorStreamTest, audioU161000f)
...
 
   /** check tensor config for audio */
-  EXPECT_TRUE (gst_tensor_config_validate (&g_test_data.tensor_config));
-  EXPECT_EQ (g_test_data.tensor_config.info.type, _NNS_UINT16);
-  EXPECT_EQ (g_test_data.tensor_config.info.dimension[0], 1U);
-  EXPECT_EQ (g_test_data.tensor_config.info.dimension[1], 1000U);
-  EXPECT_EQ (g_test_data.tensor_config.info.dimension[2], 1U);
-  EXPECT_EQ (g_test_data.tensor_config.info.dimension[3], 1U);
-  EXPECT_EQ (g_test_data.tensor_config.rate_n, 16000);
-  EXPECT_EQ (g_test_data.tensor_config.rate_d, 1);
```
Comment (anyj0527) [inline tests/nnstreamer_sink/unittest_sink.cc:2576]: ditto.

Comment (anyj0527) [inline tests/nnstreamer_sink/unittest_sink.cc:2576]: ditto.

File: tests/nnstreamer_sink/unittest_sink.cc [hunk 4]
```diff
@@ -2427,21 +2338,18 @@ TEST (tensorStreamTest, audioS8)
...
+  EXPECT_TRUE (gst_tensors_config_validate (&g_test_data.tensors_config));
+  EXPECT_EQ (g_test_data.tensors_config.info.info[0].type, _NNS_INT8);
+  EXPECT_EQ (g_test_data.tensors_config.info.info[0].dimension[0], 1U);
+  EXPECT_EQ (g_test_data.tensors_config.info.info[0].dimension[1], 500U);
+  EXPECT_EQ (g_test_data.tensors_config.info.info[0].dimension[2], 1U);
+  EXPECT_EQ (g_test_data.tensors_config.info.info[0].dimension[3], 1U);
+  EXPECT_EQ (g_test_data.tensors_config.rate_n, 16000U);
+  EXPECT_EQ (g_test_data.tensors_config.rate_d, 1U);
```
Comment (anyj0527) [inline tests/nnstreamer_sink/unittest_sink.cc:2352]: gtest will complain about the sign v.s. unsigned compare.
`16000U -> 16000`
`1U -> 1`

File: tests/nnstreamer_sink/unittest_sink.cc [hunk 5]
```diff
@@ -2559,21 +2461,18 @@ TEST (tensorStreamTest, audioU161000f)
...
-  EXPECT_EQ (g_test_data.tensor_config.rate_n, 16000);
-  EXPECT_EQ (g_test_data.tensor_config.rate_d, 1);
+  EXPECT_TRUE (gst_tensors_config_validate (&g_test_data.tensors_config));
+  EXPECT_EQ (g_test_data.tensors_config.info.info[0].type, _NNS_UINT16);
+  EXPECT_EQ (g_test_data.tensors_config.info.info[0].dimension[0], 1U);
+  EXPECT_EQ (g_test_data.tensors_config.info.info[0].dimension[1], 1000U);
+  EXPECT_EQ (g_test_data.tensors_config.info.info[0].dimension[2], 1U);
+  EXPECT_EQ (g_test_data.tensors_config.info.info[0].dimension[3], 1U);
+  EXPECT_EQ (g_test_data.tensors_config.rate_n, 16000U);
+  EXPECT_EQ (g_test_data.tensors_config.rate_d, 1U);
```
Comment (anyj0527) [inline tests/nnstreamer_sink/unittest_sink.cc:2475]: gtest will complain about the sign v.s. unsigned compare.
`16000U -> 16000`
`1U -> 1`

File: gst/nnstreamer/tensor_merge/gsttensormerge.c [hunk 6]
```diff
@@ -373,50 +373,51 @@ gst_tensor_merge_sink_event (GstCollectPads * pads, GstCollectData * data,
 }
 
 /**
- * @brief Generate TensorConfig with TensorsConfig
+ * @brief Generate out TensorsConfig with in TensorsConfig
  * @param tensor_merge tensor merger
  * @param configs Tensors Config Data
```
Comment (jaeyun-jung) [inline gst/nnstreamer/tensor_merge/gsttensormerge.c:378]: 378~379 plz update description for  in_config / out_config

File: tests/nnstreamer_sink/unittest_sink.cc (hunk 4 above)
Comment (kimjh12) [reply tests/nnstreamer_sink/unittest_sink.cc]: Updated. Thx!

File: gst/nnstreamer/tensor_merge/gsttensormerge.c (hunk 6 above)
Comment (kimjh12) [reply gst/nnstreamer/tensor_merge/gsttensormerge.c]: Updated!! Thank you

File: tests/nnstreamer_sink/unittest_sink.cc [hunk 7]
```diff
@@ -1372,26 +1362,23 @@ TEST (tensorSinkTest, signalRate)
...
 
-    caps = gst_tensor_caps_from_config (&g_test_data.tensor_config);
+    caps = gst_tensors_caps_from_config (&g_test_data.tensors_config);
     structure = gst_caps_get_structure (caps, 0);
 
-    EXPECT_TRUE (gst_tensor_config_from_structure (&config, structure));
-    EXPECT_TRUE (gst_tensor_config_is_equal (&config, &g_test_data.tensor_config));
-    EXPECT_TRUE (gst_caps_is_equal (g_test_data.current_caps, caps));
+    EXPECT_TRUE (gst_tensors_config_from_structure (&config, structure));
+    EXPECT_TRUE (gst_tensors_config_is_equal (&config, &g_test_data.tensors_config));
```
Comment (gichan-jang) [inline tests/nnstreamer_sink/unittest_sink.cc:1378]: I don't understand intention of these lines.
Get a cap from a config, get config from the cap and compare it to the original config again?
And why do you remove 1392 line?

Comment (jaeyun-jung) [reply tests/nnstreamer_sink/unittest_sink.cc:1378]: Considered to compare configuration only, not caps.
As you mentioned, this block is just for conversion between caps and config (config to caps and caps to config).
IMO, this is old testcase to check config struct, and unnecessary now :)

**Review by jaeyun-jung — APPROVED** [review]

**Review by anyj0527 — APPROVED** [review]
LGTM ✔

File: tests/nnstreamer_sink/unittest_sink.cc (hunk 7 above)
Comment (kimjh12) [reply tests/nnstreamer_sink/unittest_sink.cc:1378]: Thanks for the comment.

**Pull request merged**